### PR TITLE
Ports Lockpicking

### PR DESCRIPTION
**Pull request by Dedwolf**
## About The Pull Request

Ports lockpicking from https://github.com/Darkrp-community/OpenKeep/pull/512
Credit goes to Cre77 as the original author of this code (slightly tweaked by me for ratwood) and the original source for the spritework.

## Why It's Good For The Game
Yes it works.
image
image

### CODE REVIEW

**Comment by Gyrandola** [comment]
I gave this a quick look. I dont know how balanced it will be in practice.
Trial by fire.

**Comment by RyszardParys** [comment]
How does it work... you click and it just takes a little bit of time to lockpick or is it some skillcheck? A minigame? What is it?

**Comment by dwasint** [comment]
> I gave this a quick look. I dont know how balanced it will be in practice. Trial by fire.

personally think its completely unbalanced as it leads to people just rushing towards high value areas with picks, currently refactoring it into a minigame on vanderlin because of this, its not fun for both sides.

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -301,6 +303,64 @@
...
+		return
+	else
+		var/lockprogress = 0
+		var/locktreshold = 100
+
+		var/mob/living/L = user
+
+		var/pickskill = user.mind.get_skill_level(/datum/skill/misc/lockpicking)
+		var/perbonus = L.STAPER/2
+		var/luckbonus = L.STALUC/4
```
Comment (Heheniy) [inline code/game/objects/structures/crates_lockers/closets.dm:324]: this gives almost no bonus

**Comment by Heheniy** [comment]
pink color of the 'the locking mechanism gives' message looks out of place

**Comment by notafurryy** [comment]
would be neat if rogues got more lockpicks, the thief cubclass atleast, also maychance make knaves be able to buy them in the hoardemaster

**Review by Gyrandola — APPROVED** [review]

**Pull request merged**